integration/feature/bom/resources/build.mill

```diff
@@ -0,0 +1,200 @@
+package build
+
+import mill._
+import mill.scalalib._
+import mill.scalalib.publish._
+
+trait TestPublishModule extends PublishModule {
+  def pomSettings = PomSettings(
+    description = artifactName(),
+    organization = "com.lihaoyi.mill-tests",
+    url = "https://github.com/com-lihaoyi/mill",
+    licenses = Seq(License.`Apache-2.0`),
+    versionControl = VersionControl.github("com-lihaoyi", "mill"),
+    developers = Nil
+  )
+  def publishVersion = "0.1.0-SNAPSHOT"
+}
+
+object bom extends Module {
+  object placeholder extends JavaModule with TestPublishModule {
+    def bomDeps = Agg(
+      ivy"com.google.cloud:libraries-bom:26.50.0"
+    )
+    def ivyDeps = Agg(
+      ivy"com.google.protobuf:protobuf-java:_"
```

Comment:
Can we leave out the `:_` and just use `ivy"com.google.protobuf:protobuf-java"`? That would make our syntax consistent with Maven and Gradle, both of which just leave out the version segment without any placeholder

Reply:
Trying that

Reply:
Things like `ivy"com.google.protobuf:protobuf-java"` are now accepted.

Note that missing version translates to the version field being `"_"`, which still acts as a placeholder for versions (that makes better error mssages IMO).

So we have
```scala
ivy"com.google.protobuf:protobuf-java" == ivy"com.google.protobuf:protobuf-java:_"
```

Reply:
I think `ivy"com.google.protobuf:protobuf-java:_"` could be arguably read as "any version will do", whereas the missing versions rather means, "version needs to be configured elsewhere and is not provided here".

Reply:
Support for `_` doesn't have to be advertized, it's only used internally (it's what I used in coursier for versions meant to be read in BOMs, and so that's what Mill passes to it).

If we want to use it for something else later on, that's still possible.

Reply:
I think we need to make sure the `_` used in coursier does not leak into Mill, which is probably not possible, since we forward coursier output in error messages and `ivyDepsTree` to name some. Out of curiosity, were there any limitations forcing you to choose `_` as a marker? Would it be possible to chnage it in coursier?

Reply:
I guess it could still be changed, but for what? I used a marker so that the dependency parsers in coursier don't need much change (compared to accepting things like `org:name`)

Reply:
For the user experience? For the consistency with Maven and Gradle?

Reply:
And for avoidance of extra work in Mill, since we already use `_` as a ubiquitous wildcard, not as a marker for absence. We only have the choise to
1. convert all dependency representation coming from coursier, which is tedious and leads to a lot of boilerplate code
2. use `_` and off-load the inconsistence to the user
3. ignore and make an inconsistent UX

Reply:
So how we want to proceed now? Since `Dep` does not fully abstract over the inner `dep: Dependency`, the version is often used as is, and the forwarder `Dep.verison` wasn't touched in this PR, so we end up with a mix-up of defined or empty or `_` versions.